OSRMTextInstructions/OSRMTextInstructions.swift

```diff
@@ -179,9 +179,16 @@ public class OSRMInstructionFormatter: Formatter {
             // Set wayName
             let name = step.names?.first
             let ref = step.codes?.first
+            var isMotorWay = false
             
-            if let name = name, let ref = ref, name != ref {
+            if let roadClasses = roadClasses, roadClasses.contains(.motorway) {
+                isMotorWay = true
+            }
+            
+            if let name = name, let ref = ref, name != ref, !isMotorWay {
                 wayName = modifyValueByKey != nil ? "\(modifyValueByKey!(.wayName, name)) (\(modifyValueByKey!(.wayName, ref)))" : "\(name) (\(ref))"
+            } else if let ref = ref, isMotorWay, let decimalRange = ref.rangeOfCharacter(from: .decimalDigits), !decimalRange.isEmpty {
```

Comment:
@1ec5 tried to do 

```swift
 } else if let ref = ref, isMotorWay, let decimalRange = ref.rangeOfCharacter(from: .decimalDigits), !decimalRange.isEmpty ||  name == nil {
```

But I get a test failure with this logic. Am I using `||` correctly here?